test/controllers/update/overseas.entity.presenter.controller.spec.ts

```diff
@@ -0,0 +1,199 @@
+      expect(resp.text).not.toContain(ErrorMessages.FULL_NAME);
+    });
+
+    test("renders the current page with INVALID_CHARACTERS error message for full name and email", async () => {
+      const resp = await request(app)
+        .post(OVERSEAS_ENTITY_PRESENTER_URL)
+        .send(PRESENTER_WITH_INVALID_CHARACTERS_FIELDS_MOCK);
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(OVERSEAS_ENTITY_PRESENTER_PAGE_TITLE);
+      expect(resp.text).toContain(ErrorMessages.FULL_NAME_INVALID_CHARACTERS);
+      expect(resp.text).toContain(ErrorMessages.EMAIL_INVALID_FORMAT);
+      expect(resp.text).not.toContain(ErrorMessages.MAX_EMAIL_LENGTH);
+    });
+
+    test("renders the next page and no errors are reported if email has leading and trailing spaces", async () => {
+      const resp = await request(app)
+        .post(OVERSEAS_ENTITY_PRESENTER_URL)
+        .send(PRESENTER_OBJECT_MOCK_WITH_EMAIL_CONTAINING_LEADING_AND_TRAILING_SPACES);
+
+      expect(resp.status).toEqual(302);
+      expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${OVERSEAS_ENTITY_REVIEW_URL}`);
+
+      // Additionally check that email address is trimmed before it's saved in the session
+      const data: ApplicationDataType = mockPrepareData.mock.calls[0][0];
+      expect(data["email"]).toEqual(EMAIL_ADDRESS);
+    });
+
+    test("catch error when post data from presenter page", async () => {
+      mockSetApplicationData.mockImplementationOnce( () => { throw new Error(ANY_MESSAGE_ERROR); });
+      const resp = await request(app).post(OVERSEAS_ENTITY_PRESENTER_URL).send(PRESENTER_OBJECT_MOCK);
+
+      expect(resp.status).toEqual(500);
+      expect(resp.text).toContain(SERVICE_UNAVAILABLE);
+    });
+
+    test("Test email is valid with long email address", async () => {
+      const presenter = {
+        ...PRESENTER_OBJECT_MOCK,
+        email: "vsocarroll@QQQQQQQT123465798U123456789V123456789W123456789X123456789Y123456.companieshouse.gov.uk" };
+      const resp = await request(app)
+        .post(OVERSEAS_ENTITY_PRESENTER_URL)
+        .send(presenter);
+      expect(resp.status).toEqual(302);
+      expect(resp.text).not.toContain(ErrorMessages.MAX_EMAIL_LENGTH);
+      expect(resp.text).not.toContain(ErrorMessages.EMAIL_INVALID_FORMAT);
+    });
+
+    test("Test email is valid with long email name and address", async () => {
+      const presenter = {
+        ...PRESENTER_OBJECT_MOCK,
+        email: "socarrollA123456789B132456798C123456798D123456789@T123465798U123456789V123456789W123456789X123456789Y123456.companieshouse.gov.uk" };
+      const resp = await request(app)
+        .post(OVERSEAS_ENTITY_PRESENTER_URL)
+        .send(presenter);
+      expect(resp.status).toEqual(302);
+      expect(resp.text).not.toContain(ErrorMessages.MAX_EMAIL_LENGTH);
+      expect(resp.text).not.toContain(ErrorMessages.EMAIL_INVALID_FORMAT);
+    });
+
+    test("Test email is valid with very long email name and address", async () => {
```

Comment:
Consider using more descriptive test names as it is difficult to understand what long, and very long refers to. Maybe mention the character length for the email names/addresses? Also, might be a good idea to shorten the length of the test email addresses a little bit, even if it is to show that long emails are valid.